snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -256,6 +256,10 @@ def _inject_snap(self, name):
                            is_dangerous=rev.startswith('x'),
                            is_classic=is_classic)
 
+    def _push_file(self, src, dst):
+        subprocess.check_call(['lxc', 'file', 'push',
+                              src, '{}{}'.format(self._container_name, dst)])
+
```

Comment:
Does it seem odd to anyone else that, while the parent class now has the ability to push files, only the subclass has the ability to pull them? It sorta feels like both should be promoted.

Reply:
This is a case of need versus ability - I tried to put methods where they're actually needed in the split.  Pull is only used by Cleanbuilder to get the resulting snap.

Reply:
from a design PoV I would agree with @kyrofa though, but this is also a private method.

Reply:
@kalikiana I don't mean to debate why they were in cleanbuilder at all, but not having them together (regardless of where they are) is pulling at my OCD. I say we keep them together. @sergiusens I would call them "protected" methods, where users of the class shouldn't be calling them, but subclasses can. Welcome to C++!